common/src/main/java/org/opensearch/ml/common/agent/MLAgent.java

```diff
@@ -30,10 +31,11 @@
 import java.util.Set;
 
 import static org.opensearch.core.xcontent.XContentParserUtils.ensureExpectedToken;
+import static org.opensearch.ml.common.CommonValue.TENANT_ID;
 import static org.opensearch.ml.common.utils.StringUtils.getParameterMap;
 
 @EqualsAndHashCode
-@Getter
+@Data
```

Comment:
Add this will expose all parameters' set method, that seems not necessary/safe. For example, we don't want anyone can update agent type/appType once agent created

Reply:
ok I'll add setter for tenantId, then?

Reply:
Is it safe/necessary to set tenantId after agent created ? Can we add it to constructor method ?